sql/core/src/test/scala/org/apache/spark/sql/connector/DataSourceV2SQLSuite.scala

```diff
@@ -2492,6 +2492,35 @@ class DataSourceV2SQLSuite
     }
   }
 
+  test("SPARK-34547: metadata columns are resolved last") {
```

Comment:
I'm a bit confused. We have a test `SPARK-31255: Projects data column when metadata column has the same name` in this file, which tests when column name and metadata column name have conflicts. Why does that test work?

Reply:
This is covered at the [level](https://github.com/apache/spark/blob/62737e140c7b04805726a33c392c297335db7b45/sql/catalyst/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2Relation.scala#L59) of the DatasourceV2Relation, but does not apply above it.

Reply:
Ah I see! Can we remove the hack in `DatasourceV2Relation` now?

Reply:
We can remove it from `metadataOutput`, but I don't believe we can remove it in `withMetadataColumns` due to the output containing a name collision.